pkg/schema/error.go

```diff
@@ -95,13 +97,28 @@ func NewMismatchedTypeAssertionError(foundType yamlmeta.TypeWithValues, expected
 	}
 }
 
-func NewUnexpectedKeyAssertionError(found *yamlmeta.MapItem, definition *filepos.Position) error {
-	return schemaAssertionError{
-		position: found.GetPosition(),
-		expected: fmt.Sprintf("(a key defined in map) (by %s)", definition.AsCompactString()),
-		found:    fmt.Sprintf("%v", found.Key),
-		hints:    []string{"declare data values in schema and override them in a data values document"},
+// NewUnexpectedKeyAssertionError generates a schema assertion error including the context (and hints) needed to report it to the user
+func NewUnexpectedKeyAssertionError(found *yamlmeta.MapItem, definition *filepos.Position, allowedKeys []string) error {
+	key := fmt.Sprintf("%v", found.Key)
+	err := schemaAssertionError{
+		description: "Given data value is not declared in schema",
+		position:    found.GetPosition(),
+		found:       key,
+	}
+	sort.Strings(allowedKeys)
+	switch numKeys := len(allowedKeys); {
```

Comment:
style nit: no need for `;`

Reply:
This is required given that an assignment is not an expression.

The exact error message:
> pkg/schema/error.go:107:9: expected switch expression, found assignment (missing parentheses around composite literal?)